README.md

```diff
@@ -106,6 +107,60 @@ On SUSE, need to add '--allow-unsupported' flag when executing 'modprobe' comman
 $ sudo /sbin/modprobe isgx --allow-unsupported
 ``` 
 
+#### Install the SGX Driver on a UEFI Secure Boot Enableed machine
+
+If your machine has UEFI Secure Boot enabled, you may have to follow the following steps to install.
+   * Generate the key by using openssl
+
+      For Ubuntu / Debian & Red Hat Enterprise Linux / CentOS:
+
+      ```
+      $ openssl req -new -x509 -newkey rsa:2048 -keyout MOK.priv -outform DER -out MOK.der -nodes -days 36500 -subj "/CN=isgx Signing/"
+      ```
+
+      For SUSE, refer to SUSE official Signing procedure at: [Booting a custom kernel](https://en.opensuse.org/openSUSE:UEFI#Booting_a_custom_kernel)
+
+   * Signing the isgx (Follow the distrubution matches yours):
+   
+      * For signing the isgx.ko on Ubuntu / Debian, please enter following command with root privilege:
+  
+      ```
+      $ sudo /usr/src/linux-headers-$(uname -r)/scripts/sign-file sha256 ./MOK.priv ./MOK.der $(modinfo -n isgx)
+      ```
+
+      * To sign the isgx.ko on Red Hat Enterprise Linux Server or CentOS, need to run below command with root privileges.
+
+      ```
+      $ yum install openssl perl mokutil
+      $ perl /usr/src/kernels/$(uname -r)/scripts/sign-file sha256 MOK.priv MOK.der isgx.ko
+      ```
+
+  * Check if isgx signed
+  ```
+  $tail $(modinfo -n isgx) | grep "Module signature appended"
+  ```
+
+  * Register the keys to Secure Boot
+  ```
+  $ sudo mokutil --import MOK.der
+  ```
+
+  * Reboot and follow the bluescreen to enrol the MOK, it's operated by shimx64.efi and will happens after POSTS but before system boots.[Image References](https://sourceware.org/systemtap/wiki/SecureBoot)
```

Comment:
enroll

Reply:
I had made a new pull request to avoid the conflict of files of make pull request to sgx2 branch.

Could you do it at [#107](https://github.com/intel/linux-sgx-driver/pull/107)?

Cheers.

Reply:
We really don't want to copy instructions from distros because we would have to constant monitor and sync any changes in future. People could just look for standard instructions from distros for installing OOT driver with secure boot.